builder/component/pacman.py

```diff
@@ -282,6 +298,19 @@ def init_repos(self):
 			if "priority" in repo:
 				pacman_repo.priority = repo["priority"]
 
+			# add public key url and id
+			if "publickey" in repo and "keyid" not in repo:
+				raise ArchBuilderConfigError("publickey is provided without keyid")
+
+			if "keyid" in repo and "publickey" not in repo:
+				raise ArchBuilderConfigError("keyid is provided without publickey")
```

Comment:
Should not this reject the case where publickey is not specified? Make it possible to obtain gpg from a file or from a remote server.

Reply:
Thanks for this suggestion. This use-case should work now.